src/PrestaShopBundle/ApiPlatform/Provider/QueryProvider.php

```diff
@@ -31,20 +31,24 @@
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShopBundle\ApiPlatform\Exception\NoExtraPropertiesFoundException;
 
 class QueryProvider implements ProviderInterface
 {
-    private $queryBus;
-
-    public function __construct(CommandBusInterface $queryBus)
+    public function __construct(private CommandBusInterface $queryBus)
     {
-        $this->queryBus = $queryBus;
     }
 
     public function provide(Operation $operation, array $uriVariables = [], array $context = [])
     {
-        $query = $operation->getExtraProperties()['query'];
+        $extraProperties = $operation->getExtraProperties();
+
+        $query = array_key_exists('query', $extraProperties) ? $extraProperties['query'] : null;
```

Comment:
Just for the beauty of the art. Not blocking.
```suggestion
        $query = $extraProperties['query'] ?? null;
```